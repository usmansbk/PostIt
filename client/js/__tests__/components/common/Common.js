import React from 'react';
import renderer from 'react-test-renderer';
import {
  Button,
  Fab,
  Footer,
  Icon,
  InputField,
  Loader,
  Logo
} from '../../../src/components/common';

describe('Common', () => {
  const testComponent = Component => {
    const component = renderer.create(<Component />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  }

  it('should render Button', () => testComponent(Button))

  it('should render Fab', () => testComponent(Fab))

  it('should render Footer', () => testComponent(Footer))

  it('should render Icon', () => testComponent(Icon))

  it('should render InputField', () => testComponent(InputField))

  it('should render Loader', () => testComponent(Loader))

  it('should render Logo', () => testComponent(Logo))
})