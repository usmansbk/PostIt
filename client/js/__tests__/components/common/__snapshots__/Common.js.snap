// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Common should render Button 1`] = `
<div
  className={undefined}
>
  <button
    className="btn waves-effect waves-light undefined undefined"
    type={undefined}
  />
</div>
`;

exports[`Common should render Fab 1`] = `
<div
  className="fixed-action-btn"
  onClick={undefined}
>
  <a
    className="btn-floating btn-large modal-trigger undefined"
    href={undefined}
  >
    <i
      className="material-icons white-text"
    />
  </a>
</div>
`;

exports[`Common should render Footer 1`] = `
<footer>
  <div
    className="footer-copyright"
  >
    <small
      className="grey-text"
    >
      <p
        className={undefined}
      >
        © 
        2018
         Andela 
        <b>
          ·
        </b>
         MIT License 
        <b>
          ·
        </b>
         GitHub 
        <b>
          ·
        </b>
         facebook
      </p>
    </small>
  </div>
</footer>
`;

exports[`Common should render Icon 1`] = `
<i
  className="material-icons undefined"
/>
`;

exports[`Common should render InputField 1`] = `
<div
  className={undefined}
>
  <label
    htmlFor={undefined}
  />
  <input
    defaultValue={undefined}
    id={undefined}
    name={undefined}
    onChange={undefined}
    placeholder={undefined}
    required={undefined}
    type={undefined}
    value={undefined}
  />
</div>
`;

exports[`Common should render Loader 1`] = `
<div
  className="preloader-wrapper small active"
>
  <div
    className="spinner-layer spinner-green-only"
  >
    <div
      className="circle-clipper left"
    >
      <div
        className="circle"
      />
    </div>
    <div
      className="gap-patch"
    >
      <div
        className="circle"
      />
    </div>
    <div
      className="circle-clipper right"
    >
      <div
        className="circle"
      />
    </div>
  </div>
</div>
`;

exports[`Common should render Logo 1`] = `
<span
  className="grey-text text-darken-1 hide-on-med-and-down"
  href={undefined}
  id="brand-logo"
>
  PostIt
</span>
`;
